<!DOCTYPE html>
<html>
<body>

<h2>JavaScript Errors</h2>

<p>You cannot use the value of a non-existing variable:</p>

<p id="demo"></p>

<script>
let x = 5;
try {
  x = y + 1;
}
catch(err) {
  document.getElementById("demo").innerHTML = err.name;
}
</script>

</body>
</html>






<!-- -->
<!-- -->
<!--

JavaScript Errors - Throw and Try to Catch


-->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!--

Reference Error
A ReferenceError is thrown if you use (reference) a variable that has not been declared:



-->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!--

Error Object Properties


Property                Description

name                    Sets or returns an error name
message                 Sets or returns an error message (a string)


-->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!--


Error Name Values

Six different values can be returned by the error name property:



Error Name                  Description

EvalError                   An error has occurred in the eval() function
RangeError                  A number "out of range" has occurred
ReferenceError              An illegal reference has occurred
SyntaxError                 A syntax error has occurred
TypeError                   A type error has occurred
URIError                    An error in encodeURI() has occurred



-->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
<!-- -->
